src/FunctionApps/python/GenerateCSVs/ecr.py

```diff
@@ -1,8 +1,96 @@
-from typing import List, Any
-from GenerateCSVs.patient import PATIENT_COLUMNS
+import logging
 
-ECR_COLUMNS = PATIENT_COLUMNS + []
+from typing import List
+from GenerateCSVs.patient import PATIENT_COLUMNS, parse_patient_resource
+from GenerateCSVs.elr import extract_loinc_lab
+from GenerateCSVs.vxu import parse_immunization_resource
 
+ECR_COLUMNS = (
+    PATIENT_COLUMNS
+    + [  # Immunization columns
+        "immunizationVaccineCode",
+        "immunizationVaccineDescription",
+        "immunizationOccurrenceDateTime",
+    ]
+    + [  # Observation columns
+        "observationLoincCode",
+        "observationResult",
+        "observationEffectiveDateTime",
+    ]
+)
 
-def ecr_to_csv(bundle: dict) -> List[Any]:
-    pass
+
+def ecr_to_csv(bundle: dict) -> List[List[str]]:
+    """
+    Given a FHIR bundle containing a patient resource and one or more
+    immunization resources, create a row containing data specified by
+    ECR_COLUMNS.
+    """
+    # Initialize variables
+    patient_resources = {}
+    entries = []
+    return_rows = []
+
+    # Loop through bundle, and build:
+    #  * A dictionary to index patient resources
+    #  * A list of immunization records that should appear in the CSV
+    #  * A list of observation records that should appear in the CSV
+    for entry in bundle.get("entry"):
+        resource = entry["resource"]
+        if resource.get("resourceType") == "Patient" and len(resource.get("id")) > 0:
+            patient_resources["Patient/" + resource.get("id")] = parse_patient_resource(
+                entry
+            )
```

Comment:
I might be wrong, but doesn't Dan's `parse_patient_resource` function already have safety checks around the size of the resource's `id` array?